Cabal-syntax/src/Distribution/Fields/Lexer.x

```diff
@@ -107,7 +107,7 @@ tokens :-
 <in_section> {
   $spacetab+   ; --TODO: don't allow tab as leading space
 
-  "--" $comment* ;
+  "--" $comment* ; -- TODO capture comments instead of deleting them
```

Comment:
I got the code for this

Reply:
It's here https://github.com/andreabedini/cabal/tree/lexer-ws-and-comments, the lexer seems to work fine but I haven't finished adapting the parser. Feel free to merge the parts you need.

Reply:
ohh, thanks! I'll do that